src/pymmcore_widgets/_shutter_widget.py

```diff
@@ -244,25 +243,24 @@ def _refresh_shutter_widget(self) -> None:
             if self.autoshutter:
                 self.autoshutter_checkbox.setEnabled(False)
         else:
-            self._close_shutter(self.shutter_device)
             if self.autoshutter:
                 self.autoshutter_checkbox.setEnabled(True)
-                self.autoshutter_checkbox.setChecked(True)
-                self.shutter_button.setEnabled(False)
+                self.autoshutter_checkbox.setChecked(self._mmc.getAutoShutter())
             else:
+                self.autoshutter_checkbox.setChecked(False)
+            if self._mmc.getShutterDevice() == self.shutter_device:
                 self.shutter_button.setEnabled(not self._mmc.getAutoShutter())
+            else:
+                self.shutter_button.setEnabled(True)
```

Comment:
@ianhi why do we need to do this comparison? isn't it better to just do

```python
self.shutter_button.setEnabled(not self._mmc.getAutoShutter())
```

since when the `autoshutter` is on all the shutter should be disabled? 

For example, if you run this code, the shutter s2 is enabled even if it should not...

```python
from pymmcore_widgets import ShuttersWidget
from pymmcore_plus import CMMCorePlus
mmc = CMMCorePlus.instance()
mmc.loadSystemConfiguration()
s1 = ShuttersWidget("White Light Shutter")
s2 = ShuttersWidget("LED Shutter")
s1.show()
s2.show()
```

Reply:
> since when the autoshutter is on all the shutter should be disabled?

I think this may not be true. To my understanding the way autoshutter works is the core will take control of the current shutter device (of which there can only ever be one) and toggle that open and closed when images are taken, while leaving other shutters in whatever state the user wants. 

So in your example I'd argue that while `s2` should indeed not be toggled open (given the naming scheme) there is no reason that the button should be disabled - i.e. if the user wanted to open it for some reason then we shouldn't prevent them from doing so.

Reply:
ok I see your point! :thumbsup: